### PR TITLE
feat: add AdamW

### PR DESCRIPTION
**Pull request by TingquanGao**
Support AdamW

### CODE REVIEW

**Review by cuicheng01 — COMMENTED** [review]
水龙的PR merge后重新修改下吧

File: ppcls/optimizer/optimizer.py [hunk 1]
```diff
@@ -120,13 +120,64 @@ def __init__(self,
...
+                 beta1=0.9,
+                 beta2=0.999,
+                 epsilon=1e-8,
+                 weight_decay=None,
+                 multi_precision=False,
+                 grad_clip=None,
+                 no_weight_decay_name=None,
+                 len_shape_1_no_weight_decay=False,
+                 **args):
+        super(AdamW, self).__init__()
```
Comment (cuicheng01) [inline ppcls/optimizer/optimizer.py]: 按py3的方式写吧

File: ppcls/optimizer/optimizer.py [hunk 2]
```diff
@@ -120,13 +120,64 @@ def __init__(self,
...
+    def __init__(self,
+                 learning_rate=0.001,
+                 beta1=0.9,
+                 beta2=0.999,
+                 epsilon=1e-8,
+                 weight_decay=None,
+                 multi_precision=False,
+                 grad_clip=None,
+                 no_weight_decay_name=None,
+                 len_shape_1_no_weight_decay=False,
```
Comment (cuicheng01) [inline ppcls/optimizer/optimizer.py]: 这个名字要不改成one_dim_param_no_weight_decay吧

File: ppcls/optimizer/optimizer.py (hunk 1 above)
Comment (TingquanGao) [reply ppcls/optimizer/optimizer.py]: done

File: ppcls/optimizer/optimizer.py (hunk 2 above)
Comment (TingquanGao) [reply ppcls/optimizer/optimizer.py]: done

**Review by cuicheng01 — DISMISSED** [review]

File: ppcls/optimizer/__init__.py [hunk 3]
```diff
@@ -41,18 +41,21 @@ def build_lr_scheduler(lr_config, epochs, step_each_epoch):
     return lr
 
 
-def build_optimizer(config, epochs, step_each_epoch, parameters=None):
+def build_optimizer(config, epochs, step_each_epoch, model=None):
```
Comment (TingquanGao) [inline ppcls/optimizer/__init__.py]: def build_optimizer(config, epochs, step_each_epoch, model):

model -> models

ditill (ruoyu)

File: ppcls/optimizer/__init__.py [hunk 4]
```diff
@@ -41,18 +41,21 @@ def build_lr_scheduler(lr_config, epochs, step_each_epoch):
...
     config = copy.deepcopy(config)
     # step1 build lr
     lr = build_lr_scheduler(config.pop('lr'), epochs, step_each_epoch)
     logger.debug("build lr ({}) success..".format(lr))
     # step2 build regularization
     if 'regularizer' in config and config['regularizer'] is not None:
+        if 'weight_decay' in config:
+            raise Exception(
```
Comment (TingquanGao) [inline ppcls/optimizer/__init__.py]: logger.error

File: ppcls/optimizer/__init__.py [hunk 5]
```diff
@@ -41,18 +41,21 @@ def build_lr_scheduler(lr_config, epochs, step_each_epoch):
...
+                "ConfigError: Only one of regularizer and weight_decay can be set in Optimizer Config"
+            )
         reg_config = config.pop('regularizer')
         reg_name = reg_config.pop('name') + 'Decay'
         reg = getattr(paddle.regularizer, reg_name)(**reg_config)
-    else:
-        reg = None
+        config["weight_decay"] = reg
     logger.debug("build regularizer ({}) success..".format(reg))
```
Comment (TingquanGao) [inline ppcls/optimizer/__init__.py]: ....logger.debug("build regularizer ({}) success..".format(reg))

File: ppcls/optimizer/__init__.py (hunk 4 above)
Comment (TingquanGao) [reply ppcls/optimizer/__init__.py]: logger.warning

**Review by cuicheng01 — APPROVED** [review]

**Pull request merged**